src/Npgsql/NpgsqlConnection.cs

```diff
@@ -1584,6 +1584,54 @@ public override DataTable GetSchema()
         public override DataTable GetSchema(string? collectionName, string?[]? restrictions)
             => NpgsqlSchema.GetSchema(this, collectionName, restrictions);
 
+        /// <summary>
+        /// Asynchronously returns the supported collections
+        /// </summary>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is None.</param>
+        /// <returns>The collection specified.</returns>
+#if NET // i.e. >= NET5_0
+        public override Task<DataTable> GetSchemaAsync(CancellationToken cancellationToken = default)
+#else
+        public Task<DataTable> GetSchemaAsync(CancellationToken cancellationToken = default)
+#endif
+            => GetSchemaAsync("MetaDataCollections", cancellationToken);
+
+        /// <summary>
+        /// Asynchronously returns the schema collection specified by the collection name.
+        /// </summary>
+        /// <param name="collectionName">The collection name.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is None.</param>
+        /// <returns>The collection specified.</returns>
+#if NET // i.e. >= NET5_0
+        public override Task<DataTable> GetSchemaAsync(string collectionName, CancellationToken cancellationToken = default)
+#else
+        public Task<DataTable> GetSchemaAsync(string collectionName, CancellationToken cancellationToken = default)
+#endif
+        {
+            if (cancellationToken.IsCancellationRequested)
+                return Task.FromCanceled<DataTable>(cancellationToken);
+
+            using (NoSynchronizationContextScope.Enter())
+                return NpgsqlSchema.GetSchemaAsync(this, collectionName, null, async: true, cancellationToken);
+        }
+
+        /// <summary>
+        /// Asynchronously returns the schema collection specified by the collection name filtered by the restrictions.
+        /// </summary>
+        /// <param name="collectionName">The collection name.</param>
+        /// <param name="restrictions">
+        /// The restriction values to filter the results.  A description of the restrictions is contained
+        /// in the Restrictions collection.
+        /// </param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is None.</param>
+        /// <returns>The collection specified.</returns>
+#if NET // i.e. >= NET5_0
+        public override Task<DataTable> GetSchemaAsync(string collectionName, string?[] restrictions, CancellationToken cancellationToken = default)
+#else
+        public Task<DataTable> GetSchemaAsync(string collectionName, string?[] restrictions, CancellationToken cancellationToken = default)
+#endif
+            => NpgsqlSchema.GetSchemaAsync(this, collectionName, restrictions, async: true, cancellationToken);
```

Comment:
This is missing the cancellation token check and NoSynchronizationContextScope. Please make the top two methods call into this one, and have the checks here - just like the sync GetSchema methods already do.